hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -675,6 +675,13 @@ class Emit[C](
         presentC(const(true))
       case False() =>
         presentC(const(false))
+      case Consume(value) => {
+        val iec = emitI(value)
+        iec.map(cb){pc =>
+          // Ignore pc, just return a 1
```

Comment:
I actually think maybe this should memoize pc, then ignore the result. That seems like a safe way to ensure it's being evaluated.

Reply:
Seconded. Still not sure that it will actually work.

I don't know if we'll still fully have the ability to put things in fields in the future, but I would recommend using `cb.memoizeField`, I think the JVM will optimize less around field stores since it's probably impossible to prove that public fields (which I think all of our fields are) will never be read.